testsuite/integration-arquillian/tests/other/console/src/main/java/org/keycloak/testsuite/console/page/clients/clientscopes/ClientScopesEvaluateForm.java

```diff
@@ -103,8 +103,9 @@ public void setAssignedClientScopes(Collection<String> scopes) {
     public void selectUser(String username) {
         // TODO: Should be probably better way how to work with the "ui-select2" component
         driver.findElement(By.id("select2-chosen-1")).click();
-        driver.findElement(By.className("select2-input")).sendKeys(username);
-        driver.findElement(By.className("select2-result-label")).click();
+        String subContainerClass = "#select2-drop:not([style*='display: none'])";
+        driver.findElement(By.cssSelector( subContainerClass + " .select2-input")).sendKeys(username);
+        driver.findElements(By.cssSelector(subContainerClass + " .select2-results li.select2-result-selectable")).get(0).click();
```

Comment:
Could you perhaps try to use similar approach to [this](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/other/console/src/main/java/org/keycloak/testsuite/console/page/clients/authorization/policy/ClientPolicyForm.java#L89)? The point is not to copy-paste code between Page Objects and also not to put so many ugly selectors directly into the methods.

Reply:
Hey, you can check out the `ClientSelectModal` class I've added. It hides away the ugly CSS selectors in `ClientScopesEvaluateForm` and `CreateClientMappersForm`.

I think you wanted something like this, the tests run just fine.